src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Builder;
+
+use Symfony\Component\Config\Definition\ArrayNode;
+use Symfony\Component\Config\Definition\BooleanNode;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\Config\Definition\EnumNode;
+use Symfony\Component\Config\Definition\FloatNode;
+use Symfony\Component\Config\Definition\IntegerNode;
+use Symfony\Component\Config\Definition\NodeInterface;
+use Symfony\Component\Config\Definition\PrototypedArrayNode;
+use Symfony\Component\Config\Definition\ScalarNode;
+use Symfony\Component\Config\Definition\VariableNode;
+
+/**
+ * Generate ConfigBuilders to help create valid config.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ConfigBuilderGenerator implements GeneratorInterface
+{
+    private $classes;
+
+    /**
+     * @return \Closure that will return the root config class
+     */
+    public function build(ConfigurationInterface $configuration, string $outputDir): \Closure
+    {
+        $this->classes = [];
+
+        $rootNode = $configuration->getConfigTreeBuilder()->buildTree();
+        $rootClass = new ClassBuilder('Symfony\\Config\\'.$rootNode->getPath(), $rootNode->getName());
```

Comment:
I don't really like this namespace:
- The name of the bundle is Duplicated. `Symfony\Config\Foo` would have been shortner and simpler.
- `Symfony\Config` already exists (Config component). It could lead to conflict (hopefully we don't have -ey- Foo/FooConfig classes in it.)
- This is confusing (One can wonder why the FooBundle config is located in `symfony/config` source.

What's about `SymfonyConfig\Foo`, `Symfony\BundleConfig\Foo`,

Reply:
> The name of the bundle is Duplicated. Symfony\Config\Foo would have been shortner and simpler.

I can agree with that. I was just lazy here =)

> Symfony\Config already exists (Config component).

The config component is doing `Symfony\Component\Config`, so we are safe. 

> What's about SymfonyConfig\Foo, Symfony\BundleConfig\Foo,

FYI the Runtime component is also using namespace `Symfony\Runtime` for "special" content